### DISCUSSION TITLE
nx590e tcp/ip module

### DISCUSSION THREAD

**Discussion by tubby1981** [tjko/nxgipd#19] [General]
Is it also possible to support the NX-590E tcp/ip module?

**Comment by Dogora** [comment]
Do you have any information on the protocol used? I can't find any details. It looks like it's proprietary, meant to talk only to their receiver, which is typical of this sort of equipment. We're lucky that the serial interface protocol is documented openly.

**Comment by lucioluxx** [comment]
I'm testing in these days NX-590E module, the automation protocol is based on tcp on 9997 port (can eventually be changed); the payload query/reply record is the same as serial protocol in binary format.

**Comment by tjko** [comment]
Interesting... If protocol is basically same, should be relatively easy to make nxgipd support that module.

**Comment by pepelillo** [comment]
hi.
also interested in this. would be possible to set a virtual comm port with socat connected to this tcp port, an then using your software?

**Comment by wbreadtn** [comment]
hi, I see that it's been a while since this thread was active, but I just bought a nx-590e and have not had much luck "talking" to it. I notice that there are several attempts to unload these cards on eBay, and I now understand why :) 
Still, if anyone can offer any insight on this thing, I would very much appreciate it!
